### PR TITLE
[fix](thirdparty) Fix the errors while building brpc

### LINKED ISSUE #11713: [Bug] (thirdparty) Some errors occur while building brpc
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

When building brpc, some errors occur. This issue was introduced by #11688.

Centos 6 + clang:
image

macOS Monterey + clang:
image

### What You Expected?

Fix these errors.

### How to Reproduce?

```shell
export DORIS_TOOLCHAIN=clang bash build-thirdparty.sh
```

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by adonis0147**
# Proposed changes

Issue Number: close #11713 

## Problem summary

Fix the errors while building brpc. 

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by yiguolei — APPROVED** [review]
LGTM

File: thirdparty/patches/brpc-1.2.0.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/tools/rpc_press/rpc_press_impl.cpp b/tools/rpc_press/rpc_press_impl.cpp
+index 8a873eff..d1ff2a1e 100644
+--- a/tools/rpc_press/rpc_press_impl.cpp
++++ b/tools/rpc_press/rpc_press_impl.cpp
+@@ -222,7 +222,7 @@ void RpcPress::sync_client() {
+     int64_t last_expected_time = butil::monotonic_time_ns();
+     const int64_t interval = (int64_t) (1000000000L / req_rate);
+     // the max tolerant delay between end_time and expected_time. 10ms or 10 intervals
+-    int64_t max_tolerant_delay = std::max(10000000L, 10 * interval);    
++    int64_t max_tolerant_delay = std::max<int64_t>(10000000L, 10 * interval);
```
Comment (morningman) [inline thirdparty/patches/brpc-1.2.0.patch:10]: Should we submit this patch to BRPC?

Comment (adonis0147) [reply thirdparty/patches/brpc-1.2.0.patch:10]: I think the answer is yes. I will submit a pull request to BRPC.

Comment (adonis0147) [reply thirdparty/patches/brpc-1.2.0.patch:10]: BRPC have fixed it in https://github.com/apache/incubator-brpc/pull/1846

**Pull request merged**